File: languages/django_models/title.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Title(AbstractElementModel):

    # Mandate
    authority = models.TextField(blank=True, null=True)
    eligibility = models.TextField(blank=True, null=True)
    grant_date = models.PositiveIntegerField(blank=True, null=True)
    revoke_date = models.PositiveIntegerField(blank=True, null=True)
    issuer = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="title_issuer")
    body = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="title_body")
    superior_title = models.ForeignKey("Title", on_delete=models.SET_NULL, blank=True, null=True, related_name="title_superior_title")
    holders = models.ManyToManyField("Character", blank=True, related_name="title_holders")
    symbols = models.ManyToManyField("Object", blank=True, related_name="title_symbols")

    # World
    status = models.TextField(blank=True, null=True)
    history = models.TextField(blank=True, null=True)
    characters = models.ManyToManyField("Character", blank=True, related_name="title_characters")
    institutions = models.ManyToManyField("Institution", blank=True, related_name="title_institutions")
    families = models.ManyToManyField("Family", blank=True, related_name="title_families")
    zones = models.ManyToManyField("Zone", blank=True, related_name="title_zones")
    locations = models.ManyToManyField("Location", blank=True, related_name="title_locations")
    objects = models.ManyToManyField("Object", blank=True, related_name="title_objects")
    constructs = models.ManyToManyField("Construct", blank=True, related_name="title_constructs")
    laws = models.ManyToManyField("Law", blank=True, related_name="title_laws")
    collectives = models.ManyToManyField("Collective", blank=True, related_name="title_collectives")
    creatures = models.ManyToManyField("Creature", blank=True, related_name="title_creatures")
    phenomena = models.ManyToManyField("Phenomenon", blank=True, related_name="title_phenomena")
    species = models.ManyToManyField("Species", blank=True, related_name="title_species")
    languages = models.ManyToManyField("Language", blank=True, related_name="title_languages")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Title.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Title : Element
{
    [JsonProperty("authority"), TextAttribute("")]
    public string authority;
    [JsonProperty("eligibility"), TextAttribute("")]
    public string eligibility;
    [JsonProperty("grant_date"), Integer(0)]
    public int grantDate;
    [JsonProperty("revoke_date"), Integer(0)]
    public int revokeDate;
    [JsonProperty("issuer"), ReferenceAttribute(typeof(Institution))]
    public string issuer;
    [JsonProperty("body"), ReferenceAttribute(typeof(Institution))]
    public string body;
    [JsonProperty("superior_title"), ReferenceAttribute(typeof(Title))]
    public string superiorTitle;
    [JsonProperty("holders"), ReferenceAttribute(typeof(Character), true)]
    public string holders;
    [JsonProperty("symbols"), ReferenceAttribute(typeof(Object), true)]
    public string symbols;
    [JsonProperty("status"), TextAttribute("")]
    public string status;
    [JsonProperty("history"), TextAttribute("")]
    public string history;
    [JsonProperty("characters"), ReferenceAttribute(typeof(Character), true)]
    public string characters;
    [JsonProperty("institutions"), ReferenceAttribute(typeof(Institution), true)]
    public string institutions;
    [JsonProperty("families"), ReferenceAttribute(typeof(Family), true)]
    public string families;
    [JsonProperty("zones"), ReferenceAttribute(typeof(Zone), true)]
    public string zones;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
    [JsonProperty("laws"), ReferenceAttribute(typeof(Law), true)]
    public string laws;
    [JsonProperty("collectives"), ReferenceAttribute(typeof(Collective), true)]
    public string collectives;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
}
